examples/combined_ODNP.py

```diff
@@ -236,39 +224,37 @@
     p.mw_off()
     time.sleep(16.0)
     p.start_log()
-    DNP_ini_time = time.time()
-    DNP_data = run_spin_echo(
-        nScans=config_dict["nScans"],
-        indirect_idx=0,
-        indirect_len=len(powers) + 1,
-        ph1_cyc=Ep_ph1_cyc,
-        adcOffset=config_dict["adc_offset"],
-        carrierFreq_MHz=config_dict["carrierFreq_MHz"],
-        nPoints=nPoints,
-        nEchoes=config_dict["nEchoes"],
-        p90_us=config_dict["p90_us"],
-        repetition_us=config_dict["repetition_us"],
-        tau_us=config_dict["tau_us"],
-        SW_kHz=config_dict["SW_kHz"],
-        indirect_fields=("start_times", "stop_times"),
-        ret_data=None,
-    )  # assume that the power axis is 1 longer than the
-    #                         "powers" array, so that we can also store the
-    #                         thermally polarized signal in this array (note
-    #                         that powers and other parameters are defined
-    #                         globally w/in the script, as this function is not
-    #                         designed to be moved outside the module
-    DNP_thermal_done = time.time()
-    time_axis_coords = DNP_data.getaxis("indirect")
-    time_axis_coords[0]["start_times"] = DNP_ini_time
-    time_axis_coords[0]["stop_times"] = DNP_thermal_done
+    DNP_data = None # initially, there is no data, and run_spin_echo knows how to deal with this
+    for j in range(thermal_scans):
+        DNP_ini_time = time.time()
+        # call B/C to run spin echo
+        DNP_data = run_spin_echo(
+            nScans=config_dict["nScans"],
+            indirect_idx=j,
+            indirect_len=len(powers) + thermal_scans,
+            adcOffset=config_dict["adc_offset"],
+            carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+            nPoints=nPoints,
+            nEchoes=config_dict["nEchoes"],
+            ph1_cyc=Ep_ph1_cyc,
+            p90_us=config_dict["p90_us"],
+            repetition_us=config_dict["repetition_us"],
+            tau_us=config_dict["tau_us"],
+            SW_kHz=config_dict["SW_kHz"],
+            indirect_fields=("start_times", "stop_times"),
+            ret_data=DNP_data,
+        )
+        DNP_thermal_done = time.time()
+        time_axis_coords[j]["start_times"] = DNP_ini_time
+        time_axis_coords[j]["stop_times"] = DNP_thermal_done
     power_settings_dBm = np.zeros_like(dB_settings)
     time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(time.time()))
     for j, this_dB in enumerate(dB_settings):
         logger.debug(
             "SETTING THIS POWER", this_dB, "(", dB_settings[j - 1], powers[j], "W)"
         )
         if j == 0:
+            # why do we dip lock at the lowest power -- I don't quite understand this
```

Comment:
why do we dip lock at the lowest power -- I don't quite understand this

I think I'm maybe just emphasizing that there is no mechanism for changing the frequency as you go up in power, but if that's the case, we already knew that, which is why we would like to implement some type of frequency lock in the server.

Reply:
It just needs to be called on the first initiation of the power control server

Reply:
OK -- we might actually want to change that -- that is a separate issue, though